File: LENGUAJES/JAVASCRIPT/Ejercicio5.js
// Condicionales IF, ELSE

function obtenerEdad() {
    while (true) {
        let entrada = prompt("Ingresa tu edad:");
        let edad = parseInt(entrada);
        
        if (!isNaN(edad) && edad >= 0) {
            return edad;
        } else {
            alert("La edad no puede ser negativa. Inténtalo de nuevo.");
        }
    }
}

function mayoriaEdad() {
    let edad = obtenerEdad();
    
    if (edad >= 18) {
        console.log("Eres mayor de edad.");
    } else {
        console.log("Eres menor de edad.");
    }
}
mayoriaEdad();

File: LENGUAJES/GO/Ejercicio5.go
//Condicionales IF, ELSE
package main

import (
	"fmt"
)

func main() {
	mayoriaEdad()
}

func obtenerEdad() int {
	var edad int
	for {
		fmt.Print("Ingresa tu edad: ")
		_, err := fmt.Scan(&edad)
		if err != nil || edad < 0 {
			fmt.Println("La edad no puede ser negativa. Inténtalo de nuevo.")
			// Limpiar el buffer de entrada
			var dummy string
			fmt.Scanln(&dummy)
		} else {
			return edad
		}
	}
}

func mayoriaEdad() {
	edad := obtenerEdad()
	
	if edad >= 18 {
		fmt.Println("Eres mayor de edad.")
	} else {
		fmt.Println("Eres menor de edad.")
	}
}